src/lib/hooks/__tests__/useLazyQueryWithPagination.test.tsx

```diff
@@ -1,143 +1,366 @@
   it("should fetch data successfully", async () => {
-    const { getByTestId, queryByTestId } = render(<TestComponent />);
-    expect(queryByTestId("next")).not.toBeInTheDocument();
-    expect(queryByTestId("prev")).not.toBeInTheDocument();
-
-    mockedResponse.mockResolvedValueOnce({
-      data: {
-        ...pageOneData,
-      },
-    });
+    const { result } = renderHook(() =>
+      useLazyQueryWithPagination(testQuery, testVariables)
+    );
+    expect(result.current[1].loading).toBe(false);
+    expect(result.current[1].data).toBe(null);
+    expect(result.current[1].error).toBe(null);
 
     await act(async () => {
-      getByTestId("fetch-button").click();
+      result.current[0]();
     });
+    await waitForLoadingStartAndStop(result);
+    expect(result.current[1].data).not.toBe(null);
+    expect(result.current[1].data.erc20.data).toHaveLength(1);
+    expect(result.current[1].error).toBe(null);
+  }, 10000);
 
-    expect(queryByTestId("loading")).toBeInTheDocument();
-
-    await wait(1000);
-
-    expect(queryByTestId("loading")).not.toBeInTheDocument();
-    expect(queryByTestId("data")).toHaveTextContent(
-      JSON.stringify(pageOneData)
+  it("should handle error correctly", async () => {
+    const { result } = renderHook(() =>
+      useLazyQueryWithPagination("", testVariables)
     );
-    expect(queryByTestId("error")).not.toBeInTheDocument();
-    expect(queryByTestId("next")).toBeInTheDocument();
-    expect(queryByTestId("prev")).not.toBeInTheDocument();
-  });
 
-  it("should handle error correctly", async () => {
-    const { getByTestId, queryByTestId } = render(<TestComponent />);
+    await act(async () => {
+      result.current[0]();
+    });
+    await waitForLoadingStartAndStop(result);
+    expect(result.current[1].error).not.toBe(null);
+    expect(result.current[1].data).toBe(null);
+  });
 
-    mockedResponse.mockResolvedValueOnce({
-      data: null,
-      errors: "Error message",
+  describe("config and callbacks", () => {
+    it("should call onCompleted callback when the query is completed", async () => {
+      const mockCompleteCallback = vi.fn();
+      const { result } = renderHook(() =>
+        useLazyQueryWithPagination(testQuery, testVariables, {
+          cache: false,
+          onCompleted: mockCompleteCallback,
+        })
+      );
+      await act(async () => {
+        result.current[0]();
+      });
+      await waitForLoadingStartAndStop(result);
+      expect(mockCompleteCallback).toHaveBeenCalledWith(result.current[1].data);
     });
 
-    await act(async () => {
-      getByTestId("fetch-button").click();
+    it("should return formatted data if dataFormatter callback is provided", async () => {
+      const mockCompleteCallback = vi.fn();
+      const mockFormatterCallback = vi.fn();
+      const mockFormattedDataString = "mockFormattedDataString";
+
+      const { result } = renderHook(() =>
+        useLazyQueryWithPagination<"string">(testQuery, testVariables, {
```

Comment:
is there any significance of passing string type in useLazyQuery<"string"> here?

Reply:
not much, added it just to test the generics.